salt/modules/boto_iam.py

```diff
@@ -153,6 +153,297 @@ def role_exists(name, region=None, key=None, keyid=None, profile=None):
+
+        salt myminion boto_iam.put_group_policy mygroup policyname policyrules
+    '''
+    group = __salt__['boto_iam.get_group'](group_name=group_name, region=region,
+                                           key=key, keyid=keyid, profile=profile)
+    if group:
+        conn = _get_conn(region, key, keyid, profile)
+        try:
+            created = conn.put_group_policy(group_name, policy_name,
+                                            policy_json)
+            if created:
+                log.info('Created policy for group {0}.'.format(group_name))
+                return True
+            msg = 'Could not create policy for group {0}'
+            log.error(msg.format(group_name))
+            return False
+        except boto.exception.BotoServerError as e:
+            log.debug(e)
+            msg = 'Failed to create policy for group {0}'
+            log.error(msg.format(group_name))
+            return False
+    else:
+        log.error('Group {0} does not exist'.format(group_name))
+        return False
+
+
+def get_group_policy(group_name, policy_name, region=None, key=None,
+                     keyid=None, profile=None):
+    '''
+    Retrieves the specified policy document for the specified group.
+
+    CLI example::
+
+        salt myminion boto_iam.get_group_policy mygroup policyname
+    '''
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        info = conn.get_group_policy(group_name, policy_name)
+        log.debug('info for group policy is : {0}'.format(info))
+        if not info:
+            return False
+        info = info.get_group_policy_response.get_group_policy_result.policy_document
+        info = _unquote(info)
+        info = json.loads(info, object_pairs_hook=odict.OrderedDict)
+        return info
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to get group {0} info .'
+        log.error(msg.format(group_name))
+        return False
+
+
+def create_login_profile(user_name, password, region=None, key=None,
+                         keyid=None, profile=None):
+    '''
+    Creates a login profile for the specified user, give the user the
+    ability to access AWS services and the AWS Management Console.
+
+    CLI example::
+
+        salt myminion boto_iam.create_login_profile user_name password
+    '''
+    user = __salt__['boto_iam.get_user'](user_name, region, key, keyid, profile)
+    if not user:
+        msg = 'Username {0} does not exist'
+        log.error(msg.format(user_name))
+        return False
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        info = conn.create_login_profile(user_name, password)
+        log.info('Created profile for user {0} .'.format(user_name))
+        return info
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        if 'Conflict' in e:
+            log.info('Profile already exists for user {0} .'.format(user_name))
+            return 'Conflict'
```

Comment:
I would suggest here either False or crash. Now you  are asking to check your output every time it is Boolean or String.